### PR TITLE
Fix extending scripts by path

### LINKED ISSUE #65827: 4.0: Cannot extend a script through the editor (create button does nothing)
### Godot version

4.0.alpha17

### System information

Windows 11

### Issue description

image

Whenever I hit the Create button on this dialog, nothing happens. This error shows up in the output:

image

### Steps to reproduce

1. Create a base scene and give it a script
2. Inherit that scene (easiest, but if you just try to extend it as-is and change the name, it still doesn't work)
3. In inherited scene, extend the given script.
4. Note the create button gives the error mentioned.

### Minimal reproduction project

[ExtendScriptBug.zip](https://github.com/godotengine/godot/files/9575270/ExtendScriptBug.zip)

Note that there are at least two scenes here where this is a problem. I've added a Temp folder that shows it a bit more clearly, but it also happens with Dishwasher, inheriting from interactable.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #65827

### CODE REVIEW

File: editor/script_create_dialog.cpp [hunk 1]
```diff
@@ -373,7 +373,7 @@ void ScriptCreateDialog::_create_new() {
 	const ScriptLanguage::ScriptTemplate sinfo = _get_current_template();
 
 	String parent_class = parent_name->get_text();
-	if (!ClassDB::class_exists(parent_class) && !ScriptServer::is_global_class(parent_class)) {
+	if (!parent_name->get_text().begins_with("\"res://") && !ClassDB::class_exists(parent_class) && !ScriptServer::is_global_class(parent_class)) {
```
Comment (akien-mga) [inline editor/script_create_dialog.cpp:376]: How does it behave when using `res://script.gd` without quotes?

Comment (KoBeWi) [reply editor/script_create_dialog.cpp:376]: It's invalid
image

Comment (KoBeWi) [reply editor/script_create_dialog.cpp:376]: Ah, but it doesn't handle relative paths. Maybe using quote only is ok 🤔

Comment (akien-mga) [reply editor/script_create_dialog.cpp:376]: BTW you can reuse `parent_class` instead of `parent_name->get_text()`.

Comment (akien-mga) [reply editor/script_create_dialog.cpp:376]: `parent_class.is_quoted()`

Comment (KoBeWi) [reply editor/script_create_dialog.cpp]: Ok fixed. It relies on the path being validated properly, but it's *probably* ok.

File: editor/script_create_dialog.cpp [hunk 2]
```diff
@@ -373,7 +373,7 @@ void ScriptCreateDialog::_create_new() {
 	const ScriptLanguage::ScriptTemplate sinfo = _get_current_template();
 
 	String parent_class = parent_name->get_text();
-	if (!ClassDB::class_exists(parent_class) && !ScriptServer::is_global_class(parent_class)) {
+	if (!parent_name->get_text().begins_with("\"") && !parent_name->get_text().begins_with("'") && !ClassDB::class_exists(parent_class) && !ScriptServer::is_global_class(parent_class)) {
```
Comment (akien-mga) [inline editor/script_create_dialog.cpp:376]:
```suggestion
	if (!parent_class.is_quoted() && !ClassDB::class_exists(parent_class) && !ScriptServer::is_global_class(parent_class)) {
```

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**